packages/blocks/src/api/factory.js

```diff
@@ -321,22 +319,26 @@ export function switchToBlockType( blocks, name ) {
 	const firstBlock = blocksArray[ 0 ];
 	const sourceName = firstBlock.name;
 
-	if ( isMultiBlock && ! every( blocksArray, ( block ) => ( block.name === sourceName ) ) ) {
+	// Unless it's a `core/group` Block then check
+	// that all Blocks are of the same type otherwise
+	// we can't run a conversion
+	if ( isMultiBlock && ! name === 'core/group' && ! every( blocksArray, ( block ) => ( block.name === sourceName ) ) ) {
```

Comment:
This logic is problematic:

- I suspect you want to change `! name ===` to `name !==`. `! name` will always evaluate to a boolean, and never be equal to `'core/group'`
- Is there a way we can avoid hard-coding block names here? The block module should not have awareness of specific block implementations. There's some bad precedent in the module already, but not something we should perpetuate, or at least not without a plan for seeing it eliminated. Is it something which could be inferred from the how the block or its transforms are registered?

Reply:
> I suspect you want to change ! name === to name !==. ! name will always evaluate to a boolean, and never be equal to 'core/group'

Good catch. Thank you!

> Is there a way we can avoid hard-coding block names here? 

The issue is the method signature only provides us with the target block's `name`. I don't have anything else to go on unless I change the method signature.

I could extract a method here rather than hardcoding the name into the function?

Reply:
I thought there was an intention for multiple block types to be able to act as a group? If not, a solution might be to allow the group block to be set, the same sort of way as the default block is set:
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/index.js#L133

I also wonder about this logic `! every( blocksArray, ( block ) => ( block.name === sourceName )`. Every source block in the selection must be of the same type to allow a transformation to a destination. However, really all that's needed is for every block to be able to transform to the same destination type, not to be of the same source type. It means that for every block, the right transformation would need to be found, which might be a bit slow/convoluted. If that were changed it might be possible to drop the check against the group block.

Reply:
> I thought there was an intention for multiple block types to be able to act as a group

Currently, it appears that every source block in the selection must be of the same type to allow a transformation to a destination.

> a solution might be to allow the group block to be set, the same sort of way as the default block is set

I wasn't aware of this. It does feel like a system-wide default setting, but one we'd want users to be able to customise incase they want to use another block for grouping (yet another reason to avoid hard coding this value). 

>  all that's needed is for every block to be able to transform to the same destination type

I think this is correct and I think I know _where_ we could check this. I'll take a look.